src/components/ChecklistForm.tsx

```diff
@@ -0,0 +1,797 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { ChecklistFormProps } from '@/types/todo';
+import { createDefaultTodo, calculateTodoComplexity, calculateChecklistProgress } from '@/utils/todoUtils';
+
+export default function ChecklistForm({ onAddChecklist, parentTodoId }: ChecklistFormProps) {
+  const [title, setTitle] = useState('');
+  const [items, setItems] = useState<string[]>([]);
+  const [newItem, setNewItem] = useState('');
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [showAdvanced, setShowAdvanced] = useState(false);
+  const [validationErrors, setValidationErrors] = useState<string[]>([]);
+  const [charCount, setCharCount] = useState(0);
+  const [estimatedTime, setEstimatedTime] = useState(60);
+  const [category, setCategory] = useState('general');
+  const [tags, setTags] = useState('');
+  const [description, setDescription] = useState('');
+  const [difficulty, setDifficulty] = useState(1);
+  const [assignee, setAssignee] = useState('');
+  const [location, setLocation] = useState('');
+  const [isImportant, setIsImportant] = useState(false);
+  const [isUrgent, setIsUrgent] = useState(false);
+  const [reminderDate, setReminderDate] = useState('');
+  const [attachments, setAttachments] = useState<string[]>([]);
+  const [comments, setComments] = useState<string[]>([]);
+  const [dependencies, setDependencies] = useState<string[]>([]);
+  const [color, setColor] = useState('#3b82f6');
+  const [icon, setIcon] = useState('📋');
+  const [parentTaskId, setParentTaskId] = useState(parentTodoId);
+  const [status, setStatus] = useState('pending');
+  const [actualTime, setActualTime] = useState(0);
+  const [lastModified, setLastModified] = useState(new Date());
+  const [createdAt, setCreatedAt] = useState(new Date());
+  const [hasSubtasks, setHasSubtasks] = useState(false);
+  const [hasChecklists, setHasChecklists] = useState(true);
+  const [subtaskProgress, setSubtaskProgress] = useState(0);
+  const [checklistProgress, setChecklistProgress] = useState(0);
+  const [totalSubtasks, setTotalSubtasks] = useState(0);
+  const [completedSubtasks, setCompletedSubtasks] = useState(0);
+  const [totalChecklistItems, setTotalChecklistItems] = useState(0);
+  const [completedChecklistItems, setCompletedChecklistItems] = useState(0);
+  const [subtaskCount, setSubtaskCount] = useState(0);
+  const [checklistCount, setChecklistCount] = useState(1);
+  const [priority, setPriority] = useState(3);
+  const [dueDate, setDueDate] = useState('');
+  const [completedItems, setCompletedItems] = useState<string[]>([]);
+
+  const handleComplexChecklistValidation = () => {
+    const errors: string[] = [];
+    
+    if (title.length < 3) {
+      errors.push('Title must be at least 3 characters long');
```

Comment:
[MEDIUM] Hardcoded max text length of 200 without constant.

**Code Smell** detected

Suggested change:

```suggestion
const MAX_TEXT_LENGTH = 200;
if (text.length > MAX_TEXT_LENGTH) {
  errors.push('Title cannot exceed ' + MAX_TEXT_LENGTH + ' characters');
}
```